corehq/motech/repeaters/repeater_generators.py

```diff
@@ -636,19 +636,22 @@ def submission_user_id(self):
         return CouchUser.get_by_username(self.submission_username()).user_id
 
     def _get_configs(self, payload_doc):
-        configs = [CaseUpdateConfig.from_payload(payload_doc)]
-        extensions = payload_doc.get_subcases(CASE_INDEX_IDENTIFIER_HOST)
-        if extensions:
-            configs.extend([
-                CaseUpdateConfig.from_payload(extension_case)
-                for extension_case in extensions
-            ])
-
+        configs = self._recursive_get_configs(payload_doc)
         domains = {config.domain for config in configs}
         if len(domains) > 1:
             raise DataRegistryCaseUpdateError("Multiple updates must all be in the same domain")
         return configs
 
+    def _recursive_get_configs(self, payload_doc):
+        configs = [CaseUpdateConfig.from_payload(payload_doc)]
+        extensions = [
+            extension_case for extension_case in payload_doc.get_subcases(CASE_INDEX_IDENTIFIER_HOST)
+            if self.repeater._allowed_case_type(extension_case)
+        ]
+        for extension_case in extensions:
```

Comment:
So the hierarchy of these cases is held in the case properties and we aren't concerned with any of that when forwarding, is that right? Meaning we don't need any kind of hierarchy structure here, just a list of nested extension cases.

Reply:
One thought in relation to this: can two cases be extensions of each other? If so it'd be good to throw in a check to see if the extension is already in `configs` to avoid an infinite loop.